dmaws/stacks.py

```diff
@@ -34,14 +36,39 @@ def __init__(self, stack, stage, environment, variables):
         self.template = stack.template
         self.dependencies = stack.dependencies
 
-        self.template_body = load_file(stack.template)
+        self._template_body = None
 
         self.status = None
         self.outputs = None
         self.resources = None
 
+    @property
+    def template_body(self):
+        if self._template_body is None:
+            self._template_body = self._load_template()
+
+        return self._template_body
+
     def _build_name(self, name):
-        return template(name, {}, stage=self.stage, environment=self.environment)
+        return template(name, {},
+                        stage=self.stage, environment=self.environment)
+
+    def _load_template(self):
+        template_body = load_file(self.template)
+
+        environment_variables = [
+            p for p in self.parameters.keys() if p.startswith('EnvVar')
+        ]
+
+        def param_to_env(name):
+            s1 = re.sub('(.)([A-Z][a-z]+)', r'\1_\2', name)
+            s2 = re.sub('([a-z0-9])([A-Z])', r'\1_\2', s1)
+            return s2.upper().replace('ENV_VAR_', '')
```

Comment:
This took me a bit of thinking to understand. If what we're trying to say is "Prefix any capital letter that is not at the start of the string with an underscore" could this just be done with `re.sub('(.)([A-Z])', r'\1_\2', name)`